### PR TITLE
Run eager adam optimizer in benchmarks where possible

### PR DESCRIPTION
**Pull request by mlazos**
Runs eager Adam (instead of SGD) on all models that don't fail accuracy.

cc @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by mlazos** [comment]
Set off a benchmark run, I'll post here with the results

**Comment by albanD** [comment]
Skipping pr sanity check due to generated files.

File: benchmarks/dynamo/ci_expected_accuracy/aot_eager_huggingface_training.csv [hunk 1]
```diff
@@ -2,55 +2,55 @@ name,accuracy,graph_breaks
...
 
 
 
-AlbertForQuestionAnswering,pass,6
+AlbertForQuestionAnswering,pass,4
 
 
 
-AllenaiLongformerBase,pass,10
+AllenaiLongformerBase,pass,8
```
Comment (janeyx99) [inline benchmarks/dynamo/ci_expected_accuracy/aot_eager_huggingface_training.csv:13]: :o fewer graph breaks too? That's unexpected...

File: benchmarks/dynamo/common.py [hunk 2]
```diff
@@ -130,6 +130,109 @@ class CI(NamedTuple):
     "dlrm",
 }
 
+# These models currently fail accuracy with eager Adam optimizer
```
Comment (janeyx99) [inline benchmarks/dynamo/common.py:133]: Could you open an issue tracker and link here in the comment?

File: benchmarks/dynamo/ci_expected_accuracy/aot_eager_huggingface_training.csv (hunk 1 above)
Comment (mlazos) [reply benchmarks/dynamo/ci_expected_accuracy/aot_eager_huggingface_training.csv:13]: Yeah I think this is because of the graph breaks in the step wrapper, I wrapped the entire step in disable

Comment (janeyx99) [reply benchmarks/dynamo/ci_expected_accuracy/aot_eager_huggingface_training.csv:13]: oh nvm i forgot for a sec this is _eager_ adam

File: benchmarks/dynamo/common.py [hunk 3]
```diff
@@ -1978,12 +2084,14 @@ def run_n_iterations(self, mod, inputs):
             self.model_iter_fn(mod, inputs, collect_outputs=False)
         return self.model_iter_fn(mod, inputs, collect_outputs=True)
 
+    @torch._disable_dynamo(recursive=True)
```
Comment (janeyx99) [inline benchmarks/dynamo/common.py:2087]: Wait why is this necessary? 

And if it's cuz we're doing eager adam now, should we only disable if the optim is not SGD?

File: benchmarks/dynamo/common.py (hunk 2 above)
Comment (desertfire) [reply benchmarks/dynamo/common.py:133]: It is really hard to maintain this kind of list unless you are actively working on shrinking it. I finally got rid of the CI_SKIP list and moved to comparing against the expected result files, so it's sad to see another list like this. If we add another set of expected result files for `Adam`, I wonder how much test time that would add?

File: benchmarks/dynamo/common.py (hunk 3 above)
Comment (mlazos) [reply benchmarks/dynamo/common.py:2087]: I want to enable SGD as well. Once I enable compiled this should go away. This is a better way to disable than torch patcher

File: benchmarks/dynamo/common.py (hunk 2 above)
Comment (mlazos) [reply benchmarks/dynamo/common.py:133]: So are you suggesting doing a whole other benchmark run with Adam with a separate set of result files? I think that would add a lot of time right?

FWIW, I am planning on shrinking this list next half, because we should be running Adam on all models IMO.

**Review by desertfire — APPROVED** [review]

**Comment by mlazos** [comment]
latest run: https://github.com/pytorch/pytorch/actions/runs/7228443238
